PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/console/RunSparkScalaLivyConsoleAction.kt

```diff
@@ -22,12 +22,33 @@
 
 package com.microsoft.azure.hdinsight.spark.console
 
-import com.microsoft.azure.hdinsight.common.logger.ILogger
+import com.intellij.execution.RunManagerEx
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.microsoft.azure.hdinsight.spark.run.configuration.ArisSparkConfiguration
+import com.microsoft.azure.hdinsight.spark.run.configuration.CosmosSparkRunConfiguration
+import com.microsoft.azure.hdinsight.spark.run.configuration.LivySparkBatchJobRunConfiguration
 import org.jetbrains.plugins.scala.console.ScalaConsoleRunConfigurationFactory
 
-class RunSparkScalaLivyConsoleAction : RunSparkScalaConsoleAction(), ILogger {
+class RunSparkScalaLivyConsoleAction : RunSparkScalaConsoleAction() {
     override val consoleRunConfigurationFactory: ScalaConsoleRunConfigurationFactory
         get() = SparkScalaLivyConsoleConfigurationType().confFactory()
 
     override fun getNewSettingName(): String = "Spark Livy Interactive Session Console(Scala)"
+
+    override fun update(event: AnActionEvent) {
+        event.presentation.isEnabled = false
+        super.update(event)
+        val project = CommonDataKeys.PROJECT.getData(event.dataContext) ?: return
+        val selectedConfigSettings = RunManagerEx.getInstanceEx(project).selectedConfiguration
+        var runConfig = selectedConfigSettings?.configuration
+
```

Comment:
For example, the user sets the run config type to None but he selects an HDI config. Then we should launch livy console with HDI config rather than disabling the entry.

Reply:
Emm, code logic related to getRunConfigurationType() should go to another PR. In fact, it is for local console when no run config is selected.